### ISSUE TITLE
aglyph.compat.new_instance causes problems for components of builtin immutables (str, tuple, etc.)

### ISSUE THREAD

**Issue by mzipay** [mzipay/Aglyph#2]
In aglyph.assembler.Assembler._initialize, the aglyph.compat.new_instance helper function is used to create a new object if the initializer is a class. But if the initializer is an immutable (such as str or tuple) then the object never actually gets initialized.

For example:

```
<component id="sample" dotted-name="builtins.str">
   <init>
      <arg><str>test</str></arg>
   </init>
</component>
```

In this case, objects of "sample" will always be the empty string (i.e. never initialized to "test").

**Comment by mzipay** [comment]
Here's one possible workaround:

```
<component id="sample" dotted-name="builtins.str" factory-name="join">
   <init>
      <arg><str /></arg>
      <arg><str>test</str></arg>
   </init>
</component>
```

Use the join method, passing empty string as "self" and the string you want to initialize to as the argument. The above becomes `str.join("", "test")` and produces `"test"`.

Works the same on Python 2.7.

**Comment by mzipay** [comment]
This is fixed in release 2.1.1.
